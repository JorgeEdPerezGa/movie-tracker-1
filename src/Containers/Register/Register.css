.welcome-text {
  font-size: 32px;
  margin-top: 0;
  font-weight: 100;
  color: #676767; }

.register-container {
  background-color: #f7f7f7;
  height: 100vh;
  width: 70%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  align-content: space-around;
  margin-left: auto;
  margin-right: auto; }

.name, .email, .password {
  width: 91%;
  background: #E2E2E2;
  margin-bottom: 23px;
  height: 35px;
  font-size: 17px;
  border: none; }

.submit-button {
  margin-top: 25px;
  width: 92%;
  height: 40px;
  font-size: 18px;
  color: white;
  background: #5600ff;
  border: none;
  font-weight: 100; }
  .submit-button:hover {
    cursor: pointer; }
  .submit-button:focus {
    outline: 0 !important; }
