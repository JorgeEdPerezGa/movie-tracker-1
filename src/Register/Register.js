import React, {Component} from 'react';


export default class Register extends Component {
  constructor() {
    super();
  }


  render() {
    return(
      <div className = 'register'>
        <h2>REGISTER PAGE</h2>
        <input
          placeholder = 'name'/>
        <input
          placeholder = 'email'/>
        <input
          placeholder = 'password'/>
        <button>submit</button>
      </div>
    )
  }
}
