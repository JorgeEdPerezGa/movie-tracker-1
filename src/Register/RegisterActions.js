export const RegisterAction = (newUser) => {
  return {
    type: 'REGISTER_ACTION',
    newUser
  }
}
