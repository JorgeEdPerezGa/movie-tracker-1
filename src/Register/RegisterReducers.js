export const newUser = (store = [], action) => {
  switch (action.type) {
    case 'REGISTER_ACTION':
      return Object.assign({}, action.newUser);
    default:
      return store;
  }
}
