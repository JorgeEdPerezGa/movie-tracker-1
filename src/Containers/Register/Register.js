import React, {Component} from 'react';
import { userLogin, userRegister } from '../../API/helper';
import { connect } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import { loginSuccess} from '../../Actions/Actions';
import { RegisterAction } from '../../Actions/Actions';
import './Register.css';


class Register extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      email: '',
      password: '',
      retypePassword: '',
      signUpError: false,
      disabled: true,
      passwordError: false
    };
  }


  async handleRegister(event) {
    event.preventDefault();
    await this.createNewUser();
  }


async createNewUser() {
  if (this.state.password === this.state.retypePassword) {
    const newUserData = await userRegister(
      this.state.email,
      this.state.password,
      this.state.name
    );
    if(newUserData.status === 'success') {
    return this.logInNewUser(newUserData)
    }
  } else {
    this.setState({
      passwordError: true
    });
  }
}

async logInNewUser(newUserData) {
  if(newUserData.status === 'success') {
    const userData = await userLogin({email: this.state.email, password: this.state.password});
    console.log(userData.data);
    this.props.loginSuccess(userData);
    this.props.history.push('/')
  } else {
    this.setState({
      signUpError: true
    });
  }
}

  handleChange(input, event) {
    this.setState({
      [input]: event.target.value,
      disabled: !this.state.email || !this.state.password || !this.state.name
    });
  }

  render() {
    return (
      <div className = 'register-container'>
        <form
          onSubmit={(event) => this.handleRegister(event)}>
          {this.state.passwordError &&
            <h2>Passwords do not match, please try again.</h2>}
          {this.state.signUpError &&
            <h2>Email Already Exists, please login or use a different email!</h2>}
          <input
            type='text'
            placeholder='Name'
            onChange={(event) => this.handleChange('name', event)}
          />
          <input
            type='email'
            placeholder='Email'
            onChange={(event) => this.handleChange('email', event)}
          />
          <input
            type='password'
            placeholder='Password'
            onChange={(event) => this.handleChange('password', event)}
          />
          <input
            type='password'
            placeholder='Please Retype Password'
            onChange={(event) => this.handleChange('retypePassword', event)}
          />
          <button type='submit'
            disabled={this.state.disabled}>
              Register
          </button>
        </form>
      </div>
    );
  }
}

const mapStateToProps = (store) => ({
  user: store.user
});

const mapDispatchToProps = (dispatch) => ({
  loginSuccess: ( user ) => { dispatch(loginSuccess(user))},
  RegisterSuccess: (user, email, password) => {dispatch(RegisterAction(user, email, password))}
});

export default connect(mapStateToProps, mapDispatchToProps)(Register);
