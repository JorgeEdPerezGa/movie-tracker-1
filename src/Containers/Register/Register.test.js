import React from 'react';
import { shallow, mount } from 'enzyme';
import { Register, mapStateToProps, mapDispatchToProps } from './Register';

describe('Register Conatiner', () => {
  describe('login', () => {
    let register;

    beforeEach( () => {
  register = shallow(
    <Register
      name='Jorge'
    />);
  });

  it('should exist', () => {
    expect(register).toBeDefined();
  });

  it('should match the snapshot', () => {
  expect(register).toMatchSnapshot();
  });

  it('should have a default state', () => {
    const expectedDefault = {
      "disabled": true,
      "email": "",
      "name": "",
      "password": "",
      "passwordError": false,
      "retypePassword": "",
      "signUpError": false
    }
    expect(register.state()).toEqual(expectedDefault);
  })
  })

  describe('mapStateToProps', () => {
    let mockStore;
    let result;

    beforeAll(() => {
      mockStore = {
        login: undefined,
        user: {
          info: {
            name: "Jorge"
          }
        },
      };
      result = mapStateToProps(mockStore);
    });

    it('should pull user name from the store', () => {
      expect(result.user.info.name).toEqual(mockStore.user.info.name);
    });
  })

  describe('mapDispatchToProps', () => {
    let mockDispatch;
    let result;

    beforeEach(() => {
      mockDispatch = jest.fn();
      result = mapDispatchToProps(mockDispatch);
    });

    it('should call dispatch when loginSuccess is called', () => {
      const mockUserInfo = {mock: 'info'};

      result.loginSuccess(mockUserInfo);
      expect(mockDispatch).toHaveBeenCalled();
    });

    it('should call dispatch when setFavorites is called', () => {
      const mockUserInfo = {mock: 'info'};

      result.RegisterSuccess(mockUserInfo);
      expect(mockDispatch).toHaveBeenCalled();
    });
  });
})
