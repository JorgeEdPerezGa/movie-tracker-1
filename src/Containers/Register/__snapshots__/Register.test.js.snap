// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Conatiner login should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Register
    name="Jorge"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="welcome-text"
        >
          Get Started!
        </h2>,
        <form
          onSubmit={[Function]}
        >
          <input
            className="name"
            onChange={[Function]}
            placeholder="Name"
            type="text"
          />
          <input
            className="email"
            onChange={[Function]}
            placeholder="Email"
            type="email"
          />
          <input
            className="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
          />
          <input
            className="name"
            onChange={[Function]}
            placeholder="Please Retype Password"
            type="password"
          />
          <button
            className="submit-button"
            disabled={true}
            type="submit"
          >
            Register
          </button>
        </form>,
      ],
      "className": "register-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Get Started!",
          "className": "welcome-text",
        },
        "ref": null,
        "rendered": "Get Started!",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            false,
            <input
              className="name"
              onChange={[Function]}
              placeholder="Name"
              type="text"
            />,
            <input
              className="email"
              onChange={[Function]}
              placeholder="Email"
              type="email"
            />,
            <input
              className="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
            />,
            <input
              className="name"
              onChange={[Function]}
              placeholder="Please Retype Password"
              type="password"
            />,
            <button
              className="submit-button"
              disabled={true}
              type="submit"
            >
              Register
            </button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          false,
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "name",
              "onChange": [Function],
              "placeholder": "Name",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "email",
              "onChange": [Function],
              "placeholder": "Email",
              "type": "email",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "password",
              "onChange": [Function],
              "placeholder": "Password",
              "type": "password",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "name",
              "onChange": [Function],
              "placeholder": "Please Retype Password",
              "type": "password",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Register",
              "className": "submit-button",
              "disabled": true,
              "type": "submit",
            },
            "ref": null,
            "rendered": "Register",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="welcome-text"
          >
            Get Started!
          </h2>,
          <form
            onSubmit={[Function]}
          >
            <input
              className="name"
              onChange={[Function]}
              placeholder="Name"
              type="text"
            />
            <input
              className="email"
              onChange={[Function]}
              placeholder="Email"
              type="email"
            />
            <input
              className="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
            />
            <input
              className="name"
              onChange={[Function]}
              placeholder="Please Retype Password"
              type="password"
            />
            <button
              className="submit-button"
              disabled={true}
              type="submit"
            >
              Register
            </button>
          </form>,
        ],
        "className": "register-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Get Started!",
            "className": "welcome-text",
          },
          "ref": null,
          "rendered": "Get Started!",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              false,
              false,
              <input
                className="name"
                onChange={[Function]}
                placeholder="Name"
                type="text"
              />,
              <input
                className="email"
                onChange={[Function]}
                placeholder="Email"
                type="email"
              />,
              <input
                className="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
              />,
              <input
                className="name"
                onChange={[Function]}
                placeholder="Please Retype Password"
                type="password"
              />,
              <button
                className="submit-button"
                disabled={true}
                type="submit"
              >
                Register
              </button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            false,
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "name",
                "onChange": [Function],
                "placeholder": "Name",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "email",
                "onChange": [Function],
                "placeholder": "Email",
                "type": "email",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "password",
                "onChange": [Function],
                "placeholder": "Password",
                "type": "password",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "name",
                "onChange": [Function],
                "placeholder": "Please Retype Password",
                "type": "password",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Register",
                "className": "submit-button",
                "disabled": true,
                "type": "submit",
              },
              "ref": null,
              "rendered": "Register",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
